src/lang/uk.rs

```diff
@@ -0,0 +1,1233 @@
+            num /= bf_1000;
+        }
+
+        thousands
+    }
+
+    fn int_to_cardinal(&self, mut num: BigFloat) -> Result<String, Num2Err> {
+        // special case zero
+        if num.is_zero() {
+            return Ok(String::from(ZERO[self.declenation.index()]));
+        }
+
+        // handling negative values
+        let mut words = vec![];
+        if num.is_negative() {
+            words.push(String::from(MINUS));
+            num = -num;
+        }
+
+        // iterate over thousands
+        for (order, triplet) in self.split_thousands(num).iter().enumerate().rev() {
+            let hundreds = (triplet / 100 % 10) as usize;
+            let tens = (triplet / 10 % 10) as usize;
+            let units = (triplet % 10) as usize;
+
+            if hundreds > 0 {
+                words.push(String::from(
+                    HUNDREDS[hundreds - 1][self.declenation.index()],
+                ));
+            }
+
+            let properties = match order {
+                0 => *self, //the last group agrees with target word
+                1 => self.feminine(), //тисяча is feminite
+                _ => self.masculine(),
+            }.agreement_with_units(tens, units);
+
+            if tens == 1 {
+                words.push(format!("{}{}", TEENS_BASES[units], TEENS_FLEXIONS[self.declenation.index()]));
+            } else {
+                if tens > 1 {
+                    words.push(String::from(TENS[tens - 2][self.declenation.index()]));
+                }
+                if units == 1 || units == 2 {
+                    let props = if order==0 {self} else {&properties};
+                    words.push(String::from(
+                        GENDERED[units - 1][props.gender.index()]
+                            [props.declenation.index()],
+                    ));
+                } else if units > 0 {
+                    words.push(String::from(
+                        UNITS[units - 3][self.declenation.index()],
+                    ));
+                }
+            }
+
+            if order != 0 && triplet != &0 {
+                if order > MEGA_BASES.len() {
+                    return Err(Num2Err::CannotConvert);
+                }
+                let mega_flexion = if order==1 {
+                    THOUSAND_FLEXIONS[properties.number.index()][properties.declenation.index()]
+                } else {
+                    MEGA_FLEXIONS[properties.number.index()][properties.declenation.index()]
+                };
+                words.push(
+                    format!("{}{}",
+                    MEGA_BASES[order - 1],
+                    mega_flexion
+                ));
+            }
+        }
+
+        Ok(words.join(" "))
+    }
+
+    fn float_to_cardinal(&self, num: BigFloat) -> Result<String, Num2Err> {
+        let whole = num.int();
+        let mut numerator = num.frac().abs();
+        if numerator.is_zero() {
+            return self.int_to_cardinal(whole);
```

Comment:
this line isn't covered by the unit tests, do you mind adding a unit test that matches this case?